src/emit/instructions.rs

```diff
@@ -484,54 +486,59 @@ pub(super) fn process_instruction<'a, T: TargetRuntime<'a> + ?Sized>(
                 .build_call(bin.functions[cfg_no], &parms, "")
                 .unwrap()
                 .try_as_basic_value()
-                .left()
-                .unwrap();
+                .left();
+            //.unwrap();
 
-            let success = bin
-                .builder
-                .build_int_compare(
-                    IntPredicate::EQ,
-                    ret.into_int_value(),
-                    bin.return_values[&ReturnCode::Success],
-                    "success",
-                )
-                .unwrap();
-
-            let success_block = bin.context.append_basic_block(function, "success");
-            let bail_block = bin.context.append_basic_block(function, "bail");
-            bin.builder
-                .build_conditional_branch(success, success_block, bail_block)
-                .unwrap();
+            // Soroban doesnt have return codes, and only returns a single i64 value
```

Comment:
I was wondering if this should be enforced in sema; for Soroban, we could simply give an error diagnostic if a function returns anything else but an `int64`.

Reply:
What is reffered to here is the `wasm` functions, not Solidity functions. That is, if a Solidity function returns nothing, the `wasm` function returns an `i64` value to be XDR decoded as `Void`.

Reply:
Right, the comment could with explaining that